transport/transport.go

```diff
@@ -105,18 +102,19 @@ func (b *recvBuffer) load() {
 	if len(b.backlog) > 0 {
 		select {
 		case b.c <- b.backlog[0]:
-			b.backlog[0] = nil
+			b.backlog[0].data = nil
```

Comment:
`b.backlog[0] = recvMsg{}` is more robust if we add fields to the struct.

Also, do we ever close the channel in `recvMsg`?